src/deferred.js

```diff
@@ -321,29 +319,29 @@ jQuery.extend( {
 					}
 				};
 			},
-			progressValues, progressContexts, resolveContexts;
+			resolveContexts;
 
-		// Add listeners to Deferred subordinates; treat others as resolved
+		// Add listeners to promise-like subordinates; treat others as resolved
 		if ( length > 0 ) {
-			progressValues = new Array( length );
-			progressContexts = new Array( length );
 			resolveContexts = new Array( length );
 			for ( ; i < length; i++ ) {
```

Comment:
`length` and `i` can be merged into a single variable if you reverse the direction of this loop (e.g., `while ( i-- ) {`).